conda/plugins/manager.py

```diff
@@ -46,23 +47,44 @@ def __init__(self, project_name: str | None = None, *args, **kwargs) -> None:
             self.get_solver_backend
         )
 
-    def load_plugins(self, *plugins) -> list[str]:
+    def get_canonical_name(self, plugin: object) -> str:
+        # detect the fully qualified module name
+        prefix = "<unknown_module>"
+        if module := getmodule(plugin):
+            prefix = module.__spec__.name if module.__spec__ else module.__name__
```

Comment:
Since we've taken all of this work to avoid spoofing abilities I'm also inclined to consider any module without the `__spec__` attribute to be malformed so perhaps this is better:

```suggestion
        if (module := getmodule(plugin)) and module.__spec__:
            prefix = module.__spec__.name
```